sockeye/utils.py

```diff
@@ -58,7 +52,7 @@ def check_version(version: str):
     """
     code_version = parse_version(__version__)
     given_version = parse_version(version)
-    # TODO(fhieber): temporarily maintain backwards compatibility for 2.3.x models
+    # TODO(fhieber): maintain backwards compatibility for 2.3.x models
     if given_version[0] == '2' and given_version[1] == '3':
         logger.info(f"Code version: {__version__}")
         logger.warning(f"Given release version ({version}) does not match code version ({__version__}). "
```

Comment:
Good question. Technically people could have converted their 2.3.x models to PyTorch with Sockeye 3.0.x, and these still run with Sockeye 3.1.x. If a user tries to run with an MXNet 2.3.x model it silently do any bad things but will fail so I think this would be safe. 
What do you think?

Reply:
We could say that Sockeye 3.0 is the intermediate version that works with MXNet, PyTorch, and 2.3.x models.  Sockeye 3.1 is PyTorch-only and requires 3.0+ models.

This gives users the opportunity to convert models and run MXNet/PyTorch comparisons with 3.0.  The new PyTorch models they train will be compatible with 3.1.

Reply:
sounds good. I'll make this change